workers/unity/Packages/com.improbable.gdk.core/Worker/ViewDiff.cs

```diff
@@ -0,0 +1,305 @@
+
+        private readonly Dictionary<uint, IComponentDiffStorage> componentIdToComponentStorage =
+            new Dictionary<uint, IComponentDiffStorage>();
+
+        private readonly List<IComponentDiffStorage> componentStorageList = new List<IComponentDiffStorage>();
+
+        private readonly Dictionary<uint, Dictionary<uint, ICommandDiffStorage>> componentIdToCommandIdToStorage =
+            new Dictionary<uint, Dictionary<uint, ICommandDiffStorage>>();
+
+        private readonly List<ICommandDiffStorage> commandStorageList = new List<ICommandDiffStorage>();
+
+        private readonly WorldCommandStorage worldCommandStorage = new WorldCommandStorage();
+
+        public ViewDiff()
+        {
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                foreach (var type in assembly.GetTypes())
+                {
+                    if (typeof(IComponentDiffStorage).IsAssignableFrom(type) && !type.IsAbstract)
+                    {
+                        var instance = (IComponentDiffStorage) Activator.CreateInstance(type);
+
+                        componentStorageList.Add(instance);
+                        componentIdToComponentStorage.Add(instance.GetComponentId(), instance);
+                    }
+
+                    if (typeof(ICommandDiffStorage).IsAssignableFrom(type) && !type.IsAbstract)
+                    {
+                        var instance = (ICommandDiffStorage) Activator.CreateInstance(type);
+
+                        commandStorageList.Add(instance);
+                        if (!componentIdToCommandIdToStorage.TryGetValue(instance.GetComponentId(),
+                            out var commandIdToStorage))
+                        {
+                            commandIdToStorage = new Dictionary<uint, ICommandDiffStorage>();
+                            componentIdToCommandIdToStorage.Add(instance.GetComponentId(), commandIdToStorage);
+                        }
+
+                        commandIdToStorage.Add(instance.GetCommandId(), instance);
+                    }
+                }
+            }
+        }
+
+        public void Clean()
+        {
+            foreach (var storage in componentStorageList)
+            {
+                storage.Clear();
+            }
+
+            foreach (var storage in commandStorageList)
+            {
+                storage.Clear();
+            }
+
+            worldCommandStorage.Clear();
+            entitiesAdded.Clear();
+            entitiesRemoved.Clear();
+            logs.Clear();
+            metrics.Clear();
+            flags.Clear();
+        }
+
+        public void AddEntity(long entityId)
+        {
+            if (!entitiesRemoved.Remove(new EntityId(entityId)))
+            {
+                entitiesAdded.Add(new EntityId(entityId));
+            }
+        }
+
+        public void RemoveEntity(long entityId)
+        {
+            if (!entitiesAdded.Remove(new EntityId(entityId)))
```

Comment:
If this returns true, shouldn't we clear any components/updates/auth that have been added? Or is that resolved later

Reply:
Right now it assumes op order so that clear will always be a no op. There is an extension task to this which is to remove that assumption.